src/sync/sync_user.cpp

```diff
@@ -21,44 +21,123 @@
 #include "sync/impl/sync_metadata.hpp"
 #include "sync/sync_manager.hpp"
 #include "sync/sync_session.hpp"
+#include "sync/generic_network_transport.hpp"
+#include "../../external/json/json.hpp"
+
+#include <realm/util/base64.hpp>
 
 namespace realm {
 
+static std::string base64_decode(const std::string &in) {
+    std::string out;
+    out.resize(util::base64_decoded_size(in.size()));
+    util::base64_decode(in, &out[0], out.size());
+    return out;
+}
+
+static std::vector<std::string> split_token(const std::string& jwt) {
+    constexpr static char delimiter = '.';
+
+    std::vector<std::string> parts;
+    size_t pos = 0, start_from = 0;
+
+    while ((pos = jwt.find(delimiter, start_from)) != std::string::npos) {
+        parts.push_back(jwt.substr(start_from, pos - start_from));
+        start_from = pos + 1;
+    }
+
+    parts.push_back(jwt.substr(start_from));
+
+    if (parts.size() != 3) {
+        throw app::error::ClientError(app::error::ClientError::Code::bad_token);
+    }
+
+    return parts;
+}
+
+RealmJWT::RealmJWT(const std::string& token)
+{
+    auto parts = split_token(token);
+
+    auto json = nlohmann::json::parse(base64_decode(parts[1]));
+
+    this->token = token;
+    this->expires_at = json["exp"].get<long>();
+    this->issued_at = json["iat"].get<long>();
+
+    if (json.find("user_data") != json.end()) {
+        this->user_data = json["user_data"].get<std::map<std::string, util::Any>>();
+    }
+}
+
+ObjectSchema SyncUserProfile::schema()
+{
+    return { "user_profile",
```

Comment:
If at all possible I would strongly suggest we use appropriate types for these fields. By the looks of it it could be:
- "birthday" -> Timestamp
- "min_age" -> Int
- "max_age" -> Int

Reply:
I do not think that is a safe assumption since we are getting this data from various third party sources. Until it is sanitised by Stitch, I would prefer not to do this.